websocket_server.py

```diff
@@ -0,0 +1,269 @@
+# import asyncio
+import websockets
+import pickle, struct, time, pyaudio
+import threading
+import os
+import wave
+import textwrap
+
+import logging
+logging.basicConfig(level = logging.INFO)
+
+from collections import deque
+from dataclasses import dataclass
+
+import torch
+import numpy as np
+from websockets.sync import server
+from websockets.sync.server import serve
+from transcriber import WhisperModel
+
+
+clients = {}
+
+def recv_audio(websocket):
+    """
+    Receive audio chunks from client in an infinite loop.
+    """
+    global clients
+    client = ServeClient(websocket)
+    clients[websocket] = client
+    while True:
+        try:
+            frame_data = websocket.recv()
+            if isinstance(frame_data, str):
+                logging.info(frame_data)
+                continue
+            frame_np = np.frombuffer(frame_data, np.float32)
+            clients[websocket].add_frames(frame_np)
+            
+        except websockets.ConnectionClosedOK:
+            clients[websocket].cleanup()
+            clients.pop(websocket)
+            logging.info("Connection Closed.")
+            break
+
+
+class ServeClient:
+    RATE = 16000
+    def __init__(self, websocket, topic=None, device=None):
+        self.payload_size = struct.calcsize("Q")
+        self.data = b""
+        self.frames = b""
+        self.transcriber = WhisperModel("small.en", compute_type="float16", local_files_only=False)
+        self.timestamp_offset = 0.0
+        self.frames_np = None
+        self.frames_offset = 0.0
+        self.text = []
+        self.current_out = ''
+        self.prev_out = ''
+        self.t_start=None
+        self.exit = False
+        self.same_output_threshold = 0
+        self.show_prev_out_thresh = 5   # if pause(no output from whisper) show previous output for 5 seconds
+        self.add_pause_thresh = 3       # add a blank to segment list as a pause(no speech) for 3 seconds
+        self.transcript = []
+        self.send_last_n_segments = 2
```

Comment:
```suggestion
        self.send_last_n_segments = 10
```

To have more data to use on the client side.